source/dotnet/Library/AdaptiveCards/ParseContext.cs

```diff
@@ -13,6 +13,7 @@ public static void Initialize()
             idStack = new Stack<Tuple<string, AdaptiveInternalID, bool>>();
         }
 
+        private static Mutex mut = new Mutex();
```

Comment:
Why a mutex here instead of a simple Monitor-based lock (theC# lock statement effectively) ? A mutex is backed by kernel objects (so slower/heavier) typically to synchronize access for shared objects across processes.  Also i thought we needed to revisit the implementation here to make needing the lock unnecessary ? @Paulcam206? Not having to keep shared state would be ideal of course wrt perf and complexity but for hte short term is there a way to not even introduce any synch constructs and just turn off the usage of ParseContext?

Reply:
sure, i will replace it with a monitor.



---
In reply to: [340260783](https://github.com/microsoft/AdaptiveCards/pull/3546#discussion_r340260783) [](ancestors = 340260783)